pom.xml

```diff
@@ -1154,131 +1170,128 @@
+                      log.info('Generated the source distribution (`src.zip`) containing ' + pathByFile.size() + ' files.')
+
+                      // Short-circuit if there is no binary distribution expected
+                      if (attachmentCount == 0) {
+                          return
+                      }
+
+                      // Find auxiliary files that will go into the binary distribution
+                      pathByFile = new TreeMap()
+                      String releaseNotes = project.build.directory +
+                          '/generated-site/antora/modules/ROOT/pages/_release-notes/' +
+                          project.version + '.adoc'
+                      pathByFile.put("RELEASE-NOTES.adoc", Paths.get(releaseNotes))
+                      pathByFile.put("README.adoc", Paths.get("README.adoc"))
+                      pathByFile.put("NOTICE.txt", Paths.get("NOTICE.txt"))
+                      pathByFile.put("LICENSE.txt", Paths.get("LICENSE.txt"))
+
+                      // Find attachments that will go into the binary distribution
+                      Path projectDir = Paths.get(getProperty('maven.multiModuleProjectDirectory'))
+                      log.info("Locating attachments matching the provided pattern: ${d}{attachmentFilepathPattern}")
+                      SortedMap attachmentPathByFile = new TreeMap()
+                      Stream paths = Files.walk(Paths.get(repoDirectoryParent), 32)
+                      try {
+                          paths.forEach { path ->
+                              if (Files.isRegularFile(path)) {
+                                  String relativePath = projectDir.relativize(path).toString()
+                                  if (attachmentFilepathPattern.matcher(relativePath).matches()) {
+                                      attachmentPathByFile.put(path.getFileName().toString(), path)
+                                  }
+                              }
+                          }
+                      } catch (Exception error) {
+                          // Supplement diagnostics
+                          log.error(error)
+                          throw error
+                      } finally {
+                          paths.close()
+                      }
+
+                      StringBuilder attachments = new StringBuilder()
+                      attachmentPathByFile.eachWithIndex { entry, index ->
+                          attachments.append('\n\t[' + (index + 1) + '] ' + entry.value);
+                      }
+                      // Check if no attachments were found
+                      if (attachmentCount != attachmentPathByFile.size()) {
+                          log.error('Attachments found:' + attachments)
+                          String errorMsg = 'Error: Was expecting ' + attachmentCount + ' attachments, found ' + attachmentPathByFile.size() + '!'
+                          log.error(errorMsg)
```

Comment:
Shall we use parameters in the message instead? (There are other `log` calls where string concatenation is preferred over parameters.)

Reply:
This is not a Log4j API logger, but a [`o.a.maven.plugin.logging.Log`](https://maven.apache.org/ref/3.9.8/maven-plugin-api/apidocs/org/apache/maven/plugin/logging/Log.html). Sorry :wink:!

If you are referring to Groovy's string interpolation, I didn't manage to get it working: even if I use something like `${d}{lala}`, Maven will expand it twice into `${lala}` (assuming `${d}` expands to `$`) and then into the value of the `lala` property.
I confirmed it with a debugger.